Source/Core/VideoBackends/OGL/Render.cpp

```diff
@@ -1112,18 +1112,8 @@ void Renderer::CheckForSurfaceResize()
     return;
 
   m_main_gl_context->Update();
-  if (m_backbuffer_width == -1 && m_backbuffer_height == -1 && m_new_width != -1 &&
-      m_new_height != -1)
-  {
-    // We're on a system where we need to get the size from the toolkit
-    m_backbuffer_width = m_new_width;
-    m_backbuffer_height = m_new_height;
-  }
-  else
-  {
-    m_backbuffer_width = m_main_gl_context->GetBackBufferWidth();
-    m_backbuffer_height = m_main_gl_context->GetBackBufferHeight();
-  }
+  m_backbuffer_width = m_main_gl_context->GetBackBufferWidth();
+  m_backbuffer_height = m_main_gl_context->GetBackBufferHeight();
```

Comment:
I reverted this to handle the change entirely in the EGLWayland context instead. The way it is in your commit, `m_backbuffer_width == -1 && m_backbuffer_height == -1` fails due to the previous loop cycle having set them already